### PR TITLE
Don't drop hyperlinks when processed by the editor

### LINKED ISSUE #7455: "Toggle source view" strips hyperlinks in pasted rich-text
#### Description
Clicking "Toggle source view" strips hyperlinks in pasted rich-text.

### Steps to reproduce
1. Paste rich-text with hyperlinks into textarea.
2. Click the "Toggle source view" button.

### Environment (complete as necessary)
- Version/Git revision: 2.1.1
- Database Type: Postgres
- Database Version: 10.10
- PHP Version: 7.4.29

### Additional information/references

Copying this:
[Screenshot 2022-05-08 at 14-41-31 Do married couples need to live sexually](https://christianity.stackexchange.com/a/56224/1787)
and pasting into the textarea:
Screenshot 2022-05-08 at 14-43-21 Start new topic
gives this, after clicking "Toggle source view":
```bbc
Cappello, S.J., [i]Tractatus Canonico-Moralis de Sacramentis. Vol V. De Matrimonio[/i], p. 805 (quoted in McAuliffe, [i]Catholic Moral Teaching on the Nature and Object of Conjugal Love[/i] p. 77):
```
clicking "Toggle source view" again:
Screenshot 2022-05-08 at 14-45-34 Start new topic

### PR DESCRIPTION
**Pull request by live627**
Fix #7455

Signed-off-by: John Rayes <live627@gmail.com>

### CODE REVIEW

File: Themes/default/scripts/jquery.sceditor.smf.js [hunk 1]
```diff
@@ -767,13 +767,11 @@ sceditor.formats.bbcode.set(
 	'url', {
 		allowsEmpty: true,
 		quoteType: sceditor.BBCodeParser.QuoteType.always,
-		tags: {
-			a: {
-				'data-type': ['url']
-			}
-		},
-		format: function (element, content)
+		format(element, content)
```
Comment (Sesquipedalian) [inline Themes/default/scripts/jquery.sceditor.smf.js:770]: Why this change? It works, but it is different than anything we do elsewhere. Should we make similar changes elsewhere in this file?

Comment (live627) [reply Themes/default/scripts/jquery.sceditor.smf.js:770]: This is ES6 syntax and is a copy from my mod. I can revert it.
